contracts/PriceOracle/PriceOracle.sol

```diff
@@ -102,20 +114,42 @@ contract PriceOracle is IPriceOracle, System, Initializable, HydraChainConnector
         }
 
         // Check for suitable price
+        uint256 baseIndex = 0;
         uint256 basePrice = 0;
         uint256 totalPrice = 0;
         uint256 count = 0;
         uint256 powerSum = 0;
         for (uint256 i = 0; i < len; i++) {
-            if (prices[i].price > (basePrice * 101) / 100) {
-                // If price is outside 1% range, start a new group
-                count = 1;
-                basePrice = prices[i].price;
-                totalPrice = prices[i].price;
-                powerSum = hydraChainContract.getValidatorPower(prices[i].validator);
+            uint256 currentPriceIndex = prices[i].price;
+            if (currentPriceIndex > (basePrice * 101) / 100) {
```

Comment:
let's simplify this by having a variable that keeps the average price for the group on every step in the loop and check the 1% difference based on the latest average